### PR TITLE
[DOCS] EQL: Add wildcard support to `:` operator

### PR DESCRIPTION
**Pull request by jrodewig**
Updates docs for #65188.

### Preview
https://elasticsearch_65237.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/eql-syntax.html#eql-syntax-wildcards

### CODE REVIEW

File: docs/reference/eql/syntax.asciidoc [hunk 1]
```diff
@@ -370,6 +372,20 @@ use a regular string with the `\"` escape sequence.
...
+
+For string comparisons using the `:` operator, you can use wildcards (`*`) to
+match specific patterns:
+
+[source,eql]
+----
+field : "example*wildcard"
+field : "*example-wildcard"
+field : "example-wildcard*"
+----
```
Comment (jrodewig) [inline docs/reference/eql/syntax.asciidoc:387]: @costin Is there an escape sequence for strings containing a literal asterisk (`*`)? I tested `\*`, but that didn't seem to work.

Comment (costin) [reply docs/reference/eql/syntax.asciidoc:387]: No, not at the moment. It's something we need to discuss.

**Review by costin — APPROVED** [review]
LGTM

**Comment by jrodewig** [comment]
Thanks @costin.

**Pull request merged**